addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -77,59 +77,81 @@ function factory(dependencies) {
+                        return this._handleNotificationChannelMessage(message.payload);
+                    case 'deletion':
+                        return this._handleNotificationPartnerDeletion(message.payload);
+                    case 'inbox_needaction':
+                        return this._handleNotificationNeedaction(message.payload);
+                    case 'mark_as_read':
+                        return this._handleNotificationPartnerMarkAsRead(message.payload);
+                    case 'message_notification_update':
+                        return this._handleNotificationPartnerMessageNotificationUpdate(message.payload);
+                    case 'simple_notification':
+                        return this._handleSimpleNotification(message.payload);
+                    case 'toggle_star':
+                        return this._handleNotificationPartnerToggleStar(message.payload);
+                    case 'transient_message':
+                        return this._handleNotificationPartnerTransientMessage(message.payload);
+                    case 'typing_status':
+                        return this._handleNotificationChannelTypingStatus(message.payload);
+                    case 'unsubscribe':
                         if (id !== this.env.messaging.currentPartner.id) {
                             // ignore broadcast to other partners
                             return;
                         }
-                        return this._handleNotificationPartner(Object.assign({}, message));
+                        return this._handleNotificationPartnerUnsubscribe(message.payload);
+                    case 'author':
+                        if (id !== this.env.messaging.currentPartner.id) {
+                            // ignore broadcast to other partners
+                            return;
+                        }
+                        return this._handleNotificationPartnerAuthor(message.payload);
+                    case 'user_connection':
+                        if (id !== this.env.messaging.currentPartner.id) {
+                            // ignore broadcast to other partners
+                            return;
+                        }
+                        return this._handleNotificationPartnerUserConnection(message.payload);
+                    case 'activity_updated':
+                        if (id !== this.env.messaging.currentPartner.id) {
+                            // ignore broadcast to other partners
+                            return;
+                        }
+                        return this.env.bus.trigger('activity_updated', message.payload);
+                    case 'unfollow':
+                    case 'fold':
+                    case 'channel_invite':
+                    case 'pin_private_channel':
+                    case 'create_private_channel':
+                    case 'open_livechat':
+                    case 'join_channel':
+                    case 'channel_create':
```

Comment:
This would be better with one handler per type, as the later goal is to avoid having to send the whole channel info in all of them, so we will need more specific handlers.